### PR TITLE
pgmath: depend on awk

### LINKED ISSUE #10183: pgmath fails to build on Ubuntu (16.04 and maybe later)
@junghans @tinlun 

I tried installing `flang` in a Docker container based on Ubuntu 16.04. I tried that using both `gcc-7` and `gcc-8`. The `pgmath` dependency fails to build with segfault.

### Steps to reproduce the issue

```console
$ spack install pgmath # On Ubuntu 16.04
```

### Error Message

```
==> Building pgmath [CMakePackage]                                                                                                                [1532/30305]
==> Executing phase: 'cmake'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j32'

4 errors found in build log:
  >> 357    Segmentation fault (core dumped)
     358    lib/common/CMakeFiles/mth_z2yy.dir/build.make:63: recipe for target 'lib/common/mth_z2yy.c' failed
  >> 359    make[2]: *** [lib/common/mth_z2yy.c] Error 139
     360    make[2]: *** Deleting file 'lib/common/mth_z2yy.c'
     361    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-TbIWzt/flang-flang_20180612/spack-build'
     362    CMakeFiles/Makefile2:1044: recipe for target 'lib/common/CMakeFiles/mth_z2yy.dir/all' failed
  >> 363    make[1]: *** [lib/common/CMakeFiles/mth_z2yy.dir/all] Error 2
     364    make[1]: *** Waiting for unfinished jobs....
```
that activate the full debug output. 

### Information on your system

Ubuntu 16.04

### Reason

The default for `awk` in Ubuntu is `mawk`. If I read it right, `pgmath` uses `awk` to generate some of its own source code. `mawk` simply segfaults when running that `awk` command. Using `gawk` to run that awk command seems to be OK.

### PR DESCRIPTION
**Pull request by junghans**
Workaround for https://github.com/flang-compiler/flang/issues/434#issuecomment-378159723 and https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=901477.

Fix #10183

### CODE REVIEW

File: var/spack/repos/builtin/packages/pgmath/package.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ class Pgmath(CMakePackage):
     version('develop', branch='master')
     version('20180612', '62284e26214eaaff261a922c67f6878c')
 
+    depends_on("awk")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pgmath/package.py]: `type='build'`?

**Comment by aprokop** [comment]
I'm trying to test this out in the Ubuntu container, but I'm getting `==> Error: FetchError: All fetchers failed for pkgconf-1.5.4-2nhmpudkwqbvjcvlkmcbcnck7sukgrms`. I'll wait for a bit. But the PR looks fine to me.

**Comment by adamjstewart** [comment]
I'm also seeing problems fetching `pkgconf`. Let me take a look.

**Comment by adamjstewart** [comment]
`pkgconf` fixed in #10202.

File: var/spack/repos/builtin/packages/pgmath/package.py (hunk 1 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/pgmath/package.py]: Good catch, fixed!

**Pull request merged**